A Goa design with both a HTTP Cookie and Header response that require validation produces a client which fails to compile

Using the following Goa design:
```
package design

import (
        . "goa.design/goa/v3/dsl"
)


var _ = Service("Foo", func() {
        Method("Bar", func() {
                Result(func() {
                        Attribute("cookie", String, "A cookie")
                        Attribute("myhdr", String, "A header")
                        Required("myhdr")
                })
                HTTP(func() {
                        GET("/")
                        Response(StatusOK, func() {
                                Header("myhdr:My-Header")
                                Cookie("cookie:SID", String, func() {
                                        Format(FormatUUID)
                                })
                        })
                })
        })
})
```

Goa will generate client code which validates both the header and the cookie response from the server.  In the generated `DecodeBarResponse()` function in `gen/http/foo/client/encode_decode.go`, you will find the http.StatusOK case declares the `err error` variable twice.  Once for validating the header and a second time for validating the cookie:

```
                case http.StatusOK:
                        var (
                                myhdr string
                                err   error
                        )
                        myhdrRaw := resp.Header.Get("My-Header")
                        if myhdrRaw == "" {
                                err = goa.MergeErrors(err, goa.MissingFieldError("My-Header", "header"))
                        }
                        myhdr = myhdrRaw
                        var (
                                cookie    *string
                                cookieRaw string

                                cookies = resp.Cookies()
                                err     error
                        )
                        for _, c := range cookies {
                                switch c.Name {
                                case "SID":
                                        cookieRaw = c.Value
                                }
                        }
                        if cookieRaw != "" {
                                cookie = &cookieRaw
                        }
                        if err != nil {
                                return nil, goahttp.ErrValidationError("Foo", "Bar", err)
                        }
```

As a result, compiling the command line client will produce the error:

```
gen/http/foo/client/encode_decode.go:69:5: err redeclared in this block
        gen/http/foo/client/encode_decode.go:57:5: other declaration of err
```

--- a/http/codegen/client.go
+++ b/http/codegen/client.go
@@ -868,7 +868,9 @@ const singleResponseT = ` {{- if .ClientBody }}
 				cookies = resp.Cookies()
 		{{- if not .ClientBody }}
 			{{- if .MustValidate }}
-				err error
+				{{- if not .Headers}}
+					err error
+				{{- end }}
 			{{- end }}
 		{{- end }}
 			)

--- a/http/codegen/service_data.go
+++ b/http/codegen/service_data.go
@@ -2247,7 +2247,7 @@ func buildResponseBodyType(body, att *expr.AttributeExpr, loc *codegen.Location,
 				}
 				code, helpers, err = marshal(srcAtt, body, src, "body", svcctx, httpctx)
 				if err != nil {
-					fmt.Println(err.Error()) // TBD validate DSL so errors are not possible
+					panic(err) // bug
 				}
 				sd.ServerTransformHelpers = codegen.AppendHelpers(sd.ServerTransformHelpers, helpers)
 			}

